langchain/src/agents/chat/outputParser.ts

```diff
@@ -0,0 +1,33 @@
+import { AgentActionOutputParser } from "../../agents/types.js";
+import { AgentFinish } from "../../schema/index.js";
+import { FINAL_ANSWER_ACTION } from "./index.js";
+import { FORMAT_INSTRUCTIONS } from "./prompt.js";
+
+export class ChatAgentOutputParser extends AgentActionOutputParser {
+  async parse(text: string) {
+    if (text.includes(FINAL_ANSWER_ACTION)) {
+      const parts = text.split(FINAL_ANSWER_ACTION);
+      const output = parts[parts.length - 1].trim();
+      return { returnValues: { output }, log: text } satisfies AgentFinish;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [_, action, __] = text.split("```");
```

Comment:
I know you're just moving this from elsewhere but this can probably be improved by optionally allowing for code blocks with `json` keyword after the three backticks